### PR TITLE
✨Source Amazon Ads: migrate source to `YamlDeclarativeSource` with custom `check_connection`

### PR DESCRIPTION
**Pull request by askarpets**
## What
Resolves https://github.com/airbytehq/airbyte-internal-issues/issues/2863

## How

- Added `DeclarativeSourceAdapter`, which is inherited from `YamlDeclarativeSource` and delegates methods `check` and `streams` to the `SourceAmazonAds` class
- Replaced `SourceAmazonAds` with `DeclarativeSourceAdapter` in `run.py`
- Moved spec to `manifest.yaml` and deleted `spec.yaml`

## Recommended reading order
1. `declarative_source_adapter.py`
2. `manifest.yaml`
3. `run.py`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/declarative_source_adapter.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        return self._source.check(logger, config)
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        return self._source.streams(config)
+
+    def _validate_source(self) -> None:
+        """Skipping manifest validation as it can be incomplete when use adapter"""
+        return
+
+    def _set_adapted_methods(self) -> None:
```
Comment (girarda) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/declarative_source_adapter.py:35]: can you add a comment explaining what this is for?

**Review by girarda — APPROVED** [review]
looks great! just one request for a comment :shipit:

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/declarative_source_adapter.py (hunk 1 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/declarative_source_adapter.py:35]: Added, thanks for the suggestion!

**Comment by askarpets** [comment]
Testing summary:
    - `spec`: outputs identical
    - `check`:
        - `success`: outputs identical
        - `failure`: outputs identical (except failed rows paths)
    - `discover`: outputs identical (except the order of properties in columns with type of 'object')
    - `read`: outputs identical

@girarda could you please confirm this can be merged?

**Comment by girarda** [comment]
:shipit: @askarpets !

**Pull request merged**